xarray/backends/common.py

```diff
@@ -404,6 +411,7 @@ def __repr__(self) -> str:
             txt += f"\n  Learn more at {self.url}"
         return txt
 
+    @abstractmethod
```

Comment:
Per https://github.com/pydata/xarray/pull/7437#discussion_r1089115726, we may not want to require all of these methods. We may end up in a situation where `BackendEntrypoint`s must define one or more of `open_dataarray`, `open_dataset`, and `open_datatree`.

cc @jthielen, @keewis

Reply:
This seems pretty doable, e.g.

```python
class MyBaseClass:
    def __init__(self):
        super().__init__()

        # Check that at least one of the methods has been overridden
        if all(getattr(self.__class__, m) is getattr(MyBaseClass, m) for m in ['method1', 'method2', 'method3']):
            raise TypeError("You must implement at least one of 'method1', 'method2', 'method3'")

    def method1(self):
        pass

    def method2(self):
        pass

    def method3(self):
        pass

class MyDerivedClass(MyBaseClass):
    def method1(self):
        print("Method1 implemented")

# This will not raise an error
d = MyDerivedClass()

class MyInvalidDerivedClass(MyBaseClass):
    pass

# This will raise a TypeError
i = MyInvalidDerivedClass()
```
```bash
TypeError: You must implement at least one of 'method1', 'method2', 'method3'
```